src/backend/executor/nodeShareInputScan.c

```diff
@@ -552,11 +551,61 @@ static void sisc_lockname(char* p, int size, int share_id, const char* name)
 	}
 }
 
+
+char * joint_tmp_file_name(ShareInput_Lk_Context *lk_ctxt, char *name)
+{
+	char *tmp_file = palloc(sizeof(char) * MAXPGPATH);
+
+	strcat(tmp_file, getCurrentTempFilePath);
+	strcat(tmp_file, "/");
+	strcat(tmp_file, PG_TEMP_FILES_DIR);
+	strcat(tmp_file, "/");
```

Comment:
Please either calculate the size of tmp_file before calling strcat(),  or call strncat().
For all strcat() cases, please also modify.

Besides, You do not have the below strings as one parameter: "/" PG_TEMP_FILES_DIR "/"